applications/nrf5340_audio/src/audio_module/src/aobj_api.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright(c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _AOBJ_API_H_
+#define _AOBJ_API_H_
+
+#include <zephyr/kernel.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+/**
+ * @brief Value to define the data type carried by the block.
+ */
+enum aobj_type {
+	/* The audio data block type is undefined */
+	AOBJ_TYPE_UNDEFINED = 0,
+
+	/* The audio data block is raw PCM */
+	AOBJ_TYPE_PCM,
+
+	/* The audio data block is coded in LC3 */
+	AOBJ_TYPE_LC3
+};
+
+/**
+ * @brief Value to define the PCM interleaving type.
+ */
+enum aobj_interleaved {
+	/* The interleving is undefined */
+	AOBJ_INTERLEAVE_UNDEFINED = 0,
+
+	/* Audio samples are deinterleaved within the buffer */
+	AOBJ_DEINTERLEAVED,
+
+	/* Audio samples are interleaved within the buffer */
+	AOBJ_INTERLEAVED
+};
+
+/**
+ * @brief  A structure describing the PCM data format
+ */
+struct aobj_format {
+	/* The PCM sample rate */
+	uint32_t sample_rate;
+
+	/* Number of valid bits for a sample */
+	uint8_t bits_per_sample;
+
+	/* Number of bits used to carry a sample of size bits_per_sample*/
```

Comment:
Let's say we have a 24 bit sample in a 32 bit word, then:
bits_per_sample = 24
carrier_size = 32

This would make more sense if we still had the alignment and sign fields.

Reply:
Yes, this is the number of buts that a sample is carried in, e.g., sample size could be 24 bits but it is carried in a 32 bit word. This would make more sense if we still had alignment and sign. These all come from issues with various I2S parts, how decoders generate their output, what format encoders accept, etc.